runtime/src/qio/qio_formatted.c

```diff
@@ -360,6 +360,68 @@ qioerr _append_char(char* restrict * restrict buf, size_t* restrict buf_len, siz
 
   return 0;
 }
+qioerr read_entire_file(qio_channel_t* restrict ch, const char* restrict* restrict out, int64_t* restrict len_out, ssize_t maxlen)
+{
+    qioerr err,err_reading_file;
+    int64_t file_length;
+    err=qio_file_length(ch->file,&file_length); //extracting file size information to allocate buffer with approriate size beforehand
+    char* ret=NULL;
+    ssize_t amt;
+    err_t errcode;
+    if(!err)
+    {
+        
+        //allocate the buffer
+        ret=qio_malloc(file_length+1);
+        ret[0]='\0';
+        err_reading_file=qio_channel_read(false, ch, ret, file_length, &amt);
+        
+    }
+    else           //in case of socket or pipe files, file size info is not readily available.This has also been handled here .
+    {
+        
+        int64_t fileLenGuess=1024;
+        ret=qio_malloc(fileLenGuess+1);
+        while(true)
```

Comment:
I'd like it to work correctly if the file size is wrong or if it changes during the read. To do that, I'd like to see a while loop like this after the read we have above.

A code outline like this:
 * get file size
 * if that failed, choose a guess for initial buffer size
 * allocate initial buffer
 * loop reading in to buffer and reallocating

Reply:
Made the necessary changes